astroquery/alma/core.py

```diff
@@ -197,44 +194,92 @@ def __init__(self):
         self._sia = None
         self._tap = None
         self._datalink = None
-        self.sia_url = None
-        self.tap_url = None
-        self.datalink_url = None
+        self._sia_url = None
+        self._tap_url = None
+        self._datalink_url = None
+
+    def service_id_auth(self, archive_host):
+        """
+        Convenience property to get the current Service ID authority.  Each ARC will have a different authority in
+        the Service IDs, so we'll need to look it up.  In the case where the Registry host has been overridden,
+        use the default value.
+
+        Parameters
+        ----------
+
+        archvie_host : str
+        The discovered ARC hostname.
+
+        Returns
+        -------
+
+        The Service ID authority.  Never None.
+        """
+        return conf.service_uri_map[archive_host] if archive_host in conf.service_uri_map else conf.default_service_id_auth
 
     @property
     def datalink(self):
         if not self._datalink:
-            base_url = self._get_dataarchive_url()
-            if base_url.endswith('/'):
-                self.datalink_url = base_url + ALMA_DATALINK_PATH
-            else:
-                self.datalink_url = base_url + '/' + ALMA_DATALINK_PATH
-            self._datalink = pyvo.dal.adhoc.DatalinkService(
-                baseurl=self.datalink_url)
+            self._datalink = pyvo.dal.adhoc.DatalinkService(self.datalink_url)
         return self._datalink
 
+    @property
+    def datalink_url(self):
+        if not self._datalink_url:
+            base_url_host = requests.utils.parse_url(self._get_dataarchive_url()).host
+            try:
+                self._datalink_url = commons.get_access_url(
+                    "ivo://{}{}".format(self.service_id_auth(base_url_host), conf.datalink_service_uri_path),
```

Comment:
Prefer f-strings, also elsewhere.
```suggestion
                    "ivo://{self.service_id_auth(base_url_host)}{conf.datalink_service_uri_path}"
```

Reply:
Agree with @eerovaher , but I added the leading `f` and trailing `,`
```suggestion
                    f"ivo://{self.service_id_auth(base_url_host)}{conf.datalink_service_uri_path}",
```